src/controllers/handlers/explorer-handler.ts

```diff
@@ -19,107 +18,128 @@ const VALID_COLLECTION_TYPES = ['base-wearable', 'on-chain', 'third-party']
+      }
+      result.push({
+        type: 'on-chain',
+        ...wearable,
+        entity
+      })
+    }
+    return result
+  }
+
+  async function fetchThirdPartyWearables(thirdPartyCollectionId: string[]): Promise<MixedThirdPartyWearable[]> {
+    if (thirdPartyCollectionId.length === 0) {
+      const elements = await components.thirdPartyWearablesFetcher.fetchOwnedElements(address)
+      return elements.map(
+        (wearable: ThirdPartyWearable): MixedThirdPartyWearable => ({
+          type: 'third-party',
+          ...wearable
+        })
+      )
+    } else {
+      const elements = await Promise.all(
+        thirdPartyCollectionId.map(async (thirdPartyCollectionId) => {
+          // Strip the last part (the 6th part) if a collection contract id is specified
+          const collectionIdCleaned = thirdPartyCollectionId.split(':').slice(0, 5).join(':')
+          const urn = await parseUrn(collectionIdCleaned)
+          if (!urn || urn.type !== 'blockchain-collection-third-party-name') {
+            return []
+          }
+
+          return (await fetchThirdPartyWearablesFromThirdPartyName(components, address, urn)).map(
+            (wearable: ThirdPartyWearable): MixedThirdPartyWearable => {
+              return {
+                type: 'third-party',
                 ...wearable
-              })
-            )
+              }
+            }
           )
-        : ([] as MixedOnChainWearable[]),
-      collectionTypes.includes('third-party')
-        ? thirdPartyCollectionId.length === 0
-          ? components.thirdPartyWearablesFetcher.fetchOwnedElements(address).then((elements: ThirdPartyWearable[]) =>
-              elements.map((wearable: ThirdPartyWearable): MixedThirdPartyWearable => {
-                return {
-                  type: 'third-party',
-                  ...wearable
-                }
-              })
-            )
-          : Promise.all(
-              thirdPartyCollectionId.map(async (thirdPartyCollectionId) => {
-                // Strip the last part (the 6th part) if a collection contract id is specified
-                const collectionIdCleaned = thirdPartyCollectionId.split(':').slice(0, 5).join(':')
-                const urn = await parseUrn(collectionIdCleaned)
-                if (!urn || urn.type !== 'blockchain-collection-third-party-name') {
-                  return []
-                }
-
-                return (await fetchThirdPartyWearablesFromThirdPartyName(components, address, urn)).map(
-                  (wearable: ThirdPartyWearable): MixedThirdPartyWearable => {
-                    return {
-                      type: 'third-party',
-                      ...wearable
-                    }
-                  }
-                )
-              })
-            ).then((elements: MixedThirdPartyWearable[][]) => elements.flat(1))
-        : []
-    ])
-
-    return [...baseItems, ...nftItems, ...thirdPartyItems]
+        })
+      )
+      return elements.flat(1)
```

Comment:
only the third-party wearables need to be flattened, but I guess it achieves the same.

Reply:
I think only third-party wearables are being flattened, maybe the diff is confusing